app/src/main/kotlin/com/wire/android/ui/home/HomeViewModel.kt

```diff
@@ -74,7 +75,17 @@ class HomeViewModel @Inject constructor(
 
     fun checkRequirements() {
         viewModelScope.launch {
+            val userId = getSelf().first().id
             when {
+                shouldTriggerMigrationForUser(userId) -> {
```

Comment:
Good question, I think it makes no difference since the fallback will simply copy messages, if you think is should be done after the check for client i will change it

Reply:
No problem, for now it can stay this way, I can check that and change in my PR if needed 😉